math/shift_of_sampling_points/gen/small_overlap_type1.cpp

```diff
@@ -0,0 +1,24 @@
+#include <cstdlib>
+#include <cstdio>
+
+#include "../params.h"
+#include "random.h"
+
+int main(int, char* argv[]) {
+  long long seed = std::atoll(argv[1]);
+  auto gen = Random(seed);
+
+  const int N_SMALL = 100;
+  int n = gen.uniform<int>(1, N_SMALL);
+  int m = gen.uniform<int>(1, N_SMALL);
+  int c = int(MOD) - 1 - gen.uniform<int>(0, n - 1);
```

Comment:
The same comment as for near_max.
(If we adopt strict overlap, m should be `gen.uniform<int>(2, N_SMALL);`.)